### PR TITLE
Update Backend README

### PR DESCRIPTION
**Pull request by suryatiru**
## Problem/Purpose
- Add additional information to the backend README not covered previously

## Proposed Solution
- Included additional information for the local MySQL setup, environment setup, VSCode setup, and workflow process for implementing a specific feature

## Follow Ups
- A followup pr will be added to cover the remaining sections not included here including but not limited to AWS setup, testing, CI/CD, and diagrams.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,6 +6,88 @@ Todo: Add table of contents
...
+
+Setup:
+1. Npm install
+2. Npm run pr - creates a pr and attaches trello card description
+- Use case: when the ticket id is in the branch name or specified as an argument to the command
+3. Npm run genapispecs - generates the specs.json
+4. Npm run resetDB - deletes the data from the db tables
+5. Npm run dbtots - generates the schema and interface for each table in the db
+
+# VSCode Setup
```
Comment (suryatiru) [inline README.md]: Specify the VSCode editor is optional

File: README.md [hunk 2]
```diff
@@ -6,6 +6,88 @@ Todo: Add table of contents
...
+
+# Local MySQL database
+
+We have migrated from PlanetScale to a local MySQL database at the moment. This section will be updated once we complete the data migration over to a new db provider.
+
+# Environment variables
+- Ask a team member for the .env variables
+
+Setup:
+1. Npm install
```
Comment (suryatiru) [inline README.md]: Reminder to use the backticks to convert to code blocks

File: README.md [hunk 3]
```diff
@@ -6,6 +6,88 @@ Todo: Add table of contents
...
+- Use case: when the ticket id is in the branch name or specified as an argument to the command
+3. Npm run genapispecs - generates the specs.json
+4. Npm run resetDB - deletes the data from the db tables
+5. Npm run dbtots - generates the schema and interface for each table in the db
+
+# VSCode Setup
+
+Packages/Extensions to install when working with VSCode ->
+
+```typescript
```
Comment (suryatiru) [inline README.md:146]: Add descriptions to the extensions highlighting what they are used for

File: README.md [hunk 4]
```diff
@@ -6,6 +6,88 @@ Todo: Add table of contents
...
+2. Open the DMG file and follow the installation instructions.
+3. Open Terminal and edit your .zshrc (or .bash_profile if you're using Bash): nano ~/.zshrc
+4. Add the following line at the end of the file: export PATH="/usr/local/mysql/bin:$PATH"
+5. Save and close the file
+6. In Terminal, connect to the MySQL server: mysql -u root -p
+7. Enter your password when prompted.
+
+# MySQL setup for Windows TODO
+
+Case-insensitive table names
```
Comment (seanim0920) [inline README.md:125]: ## Install MySQL on Windows

**Note:** These instructions will guide you through the installation using Chocolatey. We recommended using Chocolatey for installing MySQL on Windows as it automatically handles dependency installation. 

### Part 1: Install Chocolatey
1. **Open an Administrative Command Prompt:**
   - Press `Win + R` to open the Run dialog.
   - Type `cmd`, then press `Ctrl + Shift + Enter` to open the command prompt with administrative privileges.

2. **Execute the Chocolatey Installation Command:**
   - Copy and paste the following command into the Command Prompt:
     ```bash
     Set-ExecutionPolicy Bypass -Scope Process -Force; [System.Net.ServicePointManager]::SecurityProtocol = [System.Net.ServicePointManager]::SecurityProtocol -bor 3072; iex ((New-Object System.Net.WebClient).DownloadString('https://chocolatey.org/install.ps1'))
     ```
   - Press `Enter` to run the command and install Chocolatey on your system.

### Part 2: Install MySQL
1. **Open an Administrative Command Prompt** (repeat the process from Step 1 if necessary).

2. **Install MySQL Using Chocolatey:**
   - Type the following command:
     ```bash
     choco install mysql
     ```
   - Press `Enter`. Chocolatey will manage the download and installation of MySQL along with its dependencies.

### Part 3: Start MySQL Server
- **Start the MySQL Service:**
  - Type the following command in the Command Prompt:
    ```bash
    net start MySQL
    ```

### Troubleshooting Common Errors
1. **Access Denied Error:**
   - If you encounter the following error:
     ```
     ERROR 1045 (28000): Access denied for user 'ODBC'@'localhost' (using password: NO)
     ```
   - Log in using:
     ```bash
     mysql -u username -p  # if your MySQL user has a password
     ```
     or
     ```bash
     mysql -u username     # if your MySQL user does not have a password
     ```

2. **MySQL Service Not Starting:**
   - **Error Message:**
     ```
     Windows could not start the MySQL service on Local Computer. Error 1067: The process terminated unexpectedly.
     ```
   - **Check the MySQL Error Log:**
     - Locate the `my.ini` file in your MySQL server directory (typically under `C:\Program Files\MySQL\MySQL Server X.X`).
     - Find the `log-error` entry to locate your error log file path.
     - Navigate to the directory containing the error log and open the file (often named something like `hostname.err`) in a text editor.

   - **Verify MySQL Data Directory:**
     - Open your MySQL configuration file (`my.ini` or `my.cnf`).
     - Ensure the `datadir` path exists and MySQL has write permissions for this directory.

   - **Re-run Initialization After Clearing Data Directory:**
     - **Stop MySQL Server:**
       ```bash
       net stop MySQL
       ```
     - **Clear the Data Directory:**
       - Locate the data directory using the `datadir` setting in the MySQL configuration file.
       - Delete all files and folders within this directory.
     - **Re-run Initialization:**
       ```bash
       mysqld --initialize
       net start MySQL
       ```

File: README.md [hunk 5]
```diff
@@ -6,6 +6,88 @@ Todo: Add table of contents
...
+
+# Local MySQL database
```
Comment (seanim0920) [inline README.md:127]: We can remove this since it doesn't contain any relevant info

File: README.md [hunk 6]
```diff
@@ -6,6 +6,88 @@ Todo: Add table of contents
...
+        "editor.defaultFormatter": "esbenp.prettier-vscode"
+    },
+    "[typescript]": {
+        "editor.defaultFormatter": "esbenp.prettier-vscode"
+    }
+}
+```
+
+# Environments
+
```
Comment (seanim0920) [inline README.md:174]: Environments are these:
("devTest"),
("stagingTest"),
("staging"),
("prod")

The current environment is set using an environment variable.

**Comment by seanim0920** [comment]
https://github.com/tifapp/FitnessProjectBackend/pull/297

**Pull request closed without merging**